src/service/tiki-service/index.ts

```diff
@@ -111,6 +112,47 @@ export class TikiService {
     this.publish.logout();
   }
 
+  checkLogin(loginType: 'GMAIL' | 'RETAILER') {
```

Comment:
This could be simplified to:

```
const startDate = this.store.sync.getStartDate()
const retailerState = this.store.retailer.get()    
const gmailState = this.store.gmail.get()
if(startDate !== undefined) return
if(loginType === 'GMAIL' && retailerState.value !== BulletState.P100) return
if(loginType === 'RETAILER' && gmailState.value !== BulletState.P100) return
this.store.sync.setStartDate()
this.store.sync.setDisconnectDate(undefined)
```

Prefer early return for readability